Libraries/LibWeb/HTML/HTMLTrackElement.cpp

```diff
@@ -156,15 +171,16 @@ void HTMLTrackElement::start_the_track_processing_model()
         return;
 
     // 4. Run the remainder of these steps in parallel, allowing whatever caused these steps to run to continue.
-    auto& realm = this->realm();
-    Platform::EventLoopPlugin::the().deferred_invoke(GC::create_function(realm.heap(), [this, &realm] {
-        m_loading = true;
-        start_the_track_processing_model_parallel_steps(realm);
+    Platform::EventLoopPlugin::the().deferred_invoke(GC::create_function(realm.heap(), [this]() {
+        start_the_track_processing_model_parallel_steps();
     }));
 }
 
-void HTMLTrackElement::start_the_track_processing_model_parallel_steps(JS::Realm& realm)
+void HTMLTrackElement::start_the_track_processing_model_parallel_steps()
 {
+    auto& realm = this->realm();
+    m_loading = true;
```

Comment:
I think it would be possible to call this more than once if `deferred_invoke()` is called multiple times before `m_loading` is set here. Setting `m_loading` before the call to `deferred_invoke()` should avoid this possibility.

This would have been an issue with the previous code as well.

Reply:
Yeah I wasn't quite sure about this thing either. It's not clear to me that we're doing this tracking correctly, but I've added a commit to set this before queuing the task.